src/journal/journal-file.c

```diff
@@ -274,6 +309,25 @@ int journal_file_set_offline(JournalFile *f, bool wait) {
         return 0;
 }
 
+/* Calls fsync on the underlying fd, respecting and maintaining f->fsync_error as needed.
+ * Not intended to be called from the offlining thread as f->fsync_error access is unserialized
+ */
+static int journal_file_fsync(JournalFile *f) {
+        assert(f);
+        assert(f->fd >= 0);
+
+        if (f->fsync_error < 0)
+                return f->fsync_error;
+
+        if (fsync(f->fd) < 0) {
+                f->fsync_error = -errno;
+
+                return -errno;
```

Comment:
not that it matters, but i would probably have shortened this to: `return f->fsync_error = -errno`